packages/protoplugin/package.json

```diff
@@ -22,17 +22,40 @@
     ".": {
       "import": "./dist/esm/index.js",
       "require": "./dist/cjs/index.js"
+    },
+    "./transpile": {
+      "import": "./dist/esm/transpile/index.js",
+      "require": "./dist/cjs/transpile/index.js"
+    }
+  },
+  "typesVersions": {
+    "*": {
+      "transpile": [
+        "./dist/cjs/transpile/index.d.ts"
+      ]
     }
   },
   "dependencies": {
-    "@bufbuild/protobuf": "2.0.0-alpha.2",
-    "@typescript/vfs": "^1.4.0",
-    "typescript": "4.5.2"
+    "@bufbuild/protobuf": "2.0.0-alpha.2"
+  },
+  "peerDependencies": {
+    "@typescript/vfs": "1.5.0",
+    "typescript": "5.4.5"
```

Comment:
We can extract `transpile` to a separate package. It can be installed by plugin authors if they need it. Then we can move away from `peerDependencies` and that will work without conflicts. It is a little bit of overhead for plugin authors, but seeing the ratio of consumer to producer I'd say it is worth it. Of course, this also means we have to maintain a new package.